Uchiwa Support

jbadiapa opened the pull request:
There are a couple of things to change.
- The sensu part (it's only needed because it is a dependency)
- The firewall dependency is not really needed here because is a dependency of sensu/server but it helps to be able to test uchiwa alone.

jbadiapa committed:
Availability Monitoring Server side

jbadiapa committed:
added rabbitmq tag

jbadiapa committed:
larsks modifications

jbadiapa committed:
Added some dependencies

mrunge committed:
performance monitoring feature

this adds support for performance monitoring for
configuring collectd node(s) and deploying/configuring
graphite-web/carbon and grafana.

paramite committed:
Avoid retry file warning

For more info see: https://github.com/ansible/ansible/issues/16365

jbadiapa committed:
Several updates
- added tox for playbooks checks
- added libselinux dependencies (removed at the end)

jbadiapa committed:
Merge remote-tracking branch 'upstream/master'

larsks committed:
add elasticsearch role

this commit adds support for installing and configuring elasticsearch

jbadiapa committed:
firewall rules added to uchiwa and rabbitmq

jbadiapa committed:
Set the correct place for some parameters

jbadiapa committed:
deleted some local files that should have not uploaded

jbadiapa committed:
empty files deleted

jbadiapa committed:
fixed some mispelling

jbadiapa committed:
added uchiwa test
copied paramters at all file to all.yml file

larsks committed:
fix errors in fluentd role

This corrects some jinja errors in the fluentd role.

jbadiapa committed:
Redis installation

jbadiapa committed:
Rabbitmq support

jbadiapa committed:
Deleted everything related to Sensu role

larsks committed:
add config for forwarding syslog messages to fluentd

this role configures fluentd to receive syslog messages, and
drops an rsyslog configuration in place to forward all syslog
messages to fluentd.

jbadiapa committed:
Redis configuration check

Set parameters to redis service, package & owner

mrunge committed:
make grafana role follow common patterns

listen for conditionals on install and service
enablement. Add a variable for the package name
and the service name.

jbadiapa committed:
Set parameters for package and service

Also deleted handler as it is not used

jbadiapa committed:
rabbitmq handler added

larsks committed:
add profile roles

these are high-level roles that contain only a "meta" directory. They
compose multiple lower-level roles into a single functional "profile".

Spredzy committed:
Collectd: Rely on the SELinux ansible facts

Currently a first task is run to determine the state of SELinux on the
machine. This task is uncesseary as Ansible provide this information as
part of the facts it gathered.

```
"ansible_selinux": {
    "config_mode": "enforcing",
    "mode": "permissive",
    "policyvers": 28,
    "status": "enabled",
    "type": "targeted"
},
```

Hence the decision to enable or not the collectd_tcp_network_connect
boolean should be based on those facts.

For clarification:

  * config_mode: is the selinux state written in /etc/selinux/config
  * mode: is the selinux state returned by getenforce

Spredzy committed:
Collectd: Standardize the way package mgmt is done

Looking at other modules in the opstools-ansible repo, it seems that
collectd isn't applying the same pattern as other.

This commit :

  * Handles the plugins as a variable
  * Understand the `manage_packages` variable
  * Understand the `manage_services` variable
  * Use a variable for its service name in its handler

Spredzy committed:
Graphite: Make graphite role follow common pattern

Make the graphite role follow common pattern. This include:

  * Use of `manage_packages` variable
  * Use of `manage_services` variable
  * Register firewalld_port

Spredzy committed:
Grafana: Enforce the grafana_port at the configuration level

Currently the grafana_port variable is enforced only at the firewall
level but not at the actual configuration file level, meaning that the
actual change of port wouldn't have the expected result.

This commit ensures the port is also changed at the configuration level.

Spredzy committed:
RabbitMQ: Allow one to specify interface and port

The playbook as it is right now allow only one to configure RabbitMQ to
listen on all interface and on port 5672.

In a real life deployment when RabbitMQ might be deployed on a server
with more than one NIC, each network having its own purpose,
making it available to all network is not recommended, hence the
operator should have a way to specify on which interface to listen on.

Port is a parameter but wasn't enforced at the configuration level, just
the firewall level. This commit allows on to be able to set the port
value at the configuration level.

Spredzy committed:
ElasticSearch: Allow one to specify on which address to listen on

This commit allows an operator to specify on which address to listen on
and on which port, using respectively `elasticsearch_interface` and
`elasticsearch_port` variable.

jbadiapa committed:
Added java installation just before the elasticsearch installation

jbadiapa committed:
Uchiwa Support

paramite commented on roles/uchiwa/templates/uchiwa.json.j2:
```diff
@@ -0,0 +1,14 @@
+       "name": "{{ uchiwa_sensu_api_server_name }}",
+       "host": "{{ sensu_api_server }}",
+       "port": {{ sensu_api_port }} 
+     }
```
Wouldn't it be better to support multiple data centers, eg. have the value of key "sensu" in defaults/main.yml, with the default value of the hash above?

jbadiapa committed:
Multiple datacenters available

larsks committed:
remove empty kibana/tasks/main.yml

mrunge committed:
Add repository configuration

This adds repository configuration for
- kibana
- elasticsearch
- grafana

jbadiapa committed:
Merged with upstream

jbadiapa committed:
profiles missing s fixed

larsks committed:
correct syntax error in rabbitmq module

The firewall port configuration in the rabbitmq role would add the
literal string `{{ rabbitmq_port }}` to the `firewall_ports` variable.
This corrects the syntax in the "Register rabbitmq firewall ports"
task.

jbadiapa committed:
Sensu Support (#18)

Sensu Support

Adds Sensu role for server and API service deployment.

larsks commented on roles/uchiwa/defaults/main.yml:
```diff
@@ -0,0 +1,11 @@
+uchiwa_file_path: /etc/uchiwa/uchiwa.json
+uchiwa_sensu_api_server_name: uchiwa
+
+datacenters: [{"name": "{{ uchiwa_sensu_api_server_name }}","host": "{{ sensu_api_server }}","port": "{{ sensu_api_port }}" }]
```
This should probably be uchiwa_datacenters, or sensu_datacenters, or something with a name prefix.

jbadiapa committed:
sensu_datacenters - proper prefix

larsks commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Create uchiwa ansible role](https://trello.com/c/5fSsq6Zp/71-create-uchiwa-ansible-role)

larsks commented:
@jbadiapa, don't merge master into your feature branch; instead `git rebase` your work on top of master:

```
git checkout master
git pull
git checkout my-feature
git rebase master
```

jbadiapa committed:
Merge remote-tracking branch 'upstream/master'

jbadiapa commented:
I think I've mess up again :(  
So I killed it
